### PR TITLE
Improve compatibility with iOS, Android. Make lib usable in browser

### PR DESCRIPTION
**Pull request by jacquarg**
Library is now usable in browser (added on window object).
Can handle quoted-printable field of vCard 2.1, sended by android vcf exports.
Add dav specifique text escape.
Change fn, n and adr behaviour to improve compatibility.

### CODE REVIEW

File: test/android.vcf [hunk 1]
```diff
@@ -1,18 +1,18 @@
 BEGIN:VCARD
 VERSION:2.1
-N:Test;Cozy;;;
-FN:Cozy Test
+N;CHARSET=UTF-8;ENCODING=QUOTED-PRINTABLE:=64=65=20=43=6F=7A=79;=54=65=73=74=C3=A9;;;
```
Comment (aenario) [inline test/android.vcf]: could you add another .vcf style instead (to be sure it still works with a non-quoted value ?

File: src/index.coffee [hunk 2]
```diff
@@ -67,9 +114,31 @@ module.exports = class VCardParser
         @storeCurrentDatapoint()
         @currentContact.datapoints.push {name, type, value}
 
+    storeCurrentContact: () ->
```
Comment (jsilvestre) [inline src/index.coffee]: useless parenthesis

File: src/index.coffee [hunk 3]
```diff
@@ -67,9 +114,31 @@ module.exports = class VCardParser
...
+        # There is two fields N and FN that does the same thing but not the
+        # same way. Some vCard have one or ther other, or both.
+        # If both are present, we keep them.
+        # If only one is present, we compute the other one.
+        if not @currentContact.n? and not @currentContact.fn?
+            console.error 'There should be at least a N field or a FN field'
+
+        if (not @currentContact.n?) or
+        @currentContact.n is '' or
+        @currentContact.n is ';;;;'
```
Comment (jsilvestre) [inline src/index.coffee]:
```coffeescript
... or @currentContact.n in ['', ';;;;']
````

File: src/index.coffee [hunk 4]
```diff
@@ -182,6 +261,51 @@ module.exports = class VCardParser
...
+        flat = streetPart
+        flat += '\n' + countryPart if countryPart isnt ''
+        return flat
+
+
+
+VCardParser.unquotePrintable = (s) ->
+    if not s?
+        s = ''
+    return utf8.decode quotedPrintable.decode s
```
Comment (jsilvestre) [inline src/index.coffee]: you can do:
```coffeescript
VCardParser.unquotePrintable = (s = '') ->
    return utf8.decode quotedPrintable.decode s
```
Or 
```coffeescript
VCardParser.unquotePrintable = (s) ->
    s = s or ''
    return utf8.decode quotedPrintable.decode s
```

**Comment by jsilvestre** [comment]
Apart from a few remarks, it's quite good. I would like to raise a red flag though, you added 400 new lines of code, and only changed on test. Can you add tests please? It's very important at this point.

**Comment by jsilvestre** [comment]
It looks good to me. Do you anything you'd like to add?

**Comment by jacquarg** [comment]
I "touch wood", but I have nothing to add. 
But the version should be updated and the new address format breaks compatibility.

**Comment by jsilvestre** [comment]
@aenario this version is breaking, can you republish to 0.2.0 or something?

**Comment by aenario** [comment]
unpublished 0.1.2 & published again as 0.2.0

**Pull request merged**